### ISSUE TITLE
(aws-eks): add service ip range setting to cluster class 

### ISSUE THREAD

**Issue by Krotki** [aws/aws-cdk#16541]
In console we can specify Service IPv4 range, same goes with `CfnCluster` class, which contains `kubernetesNetworkConfig?` property. Unfortunately it is not possible to set it via `Cluster` class. 

### Use Case

We planed to migrate out on-premis k8s kluster to aws. To do it we planned to expose pod k8s network and service k8s network to EKS and viceversa. Unfortunatly by default k8s comes with service ip range of 10.96.0.0/12, EKS comes with 10.100.0.0/16, which overlap. In order to enable this scenerio we have to change EKS service ip range.

### Proposed Solution

Actually I'm not sure how `Cluster` class works ATM, but if it uses `CfnCluster` class underneath, it is only a matter of exposing `kubernetesNetworkConfig?` property.

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by otaviomacedo** [comment]
> Actually I'm not sure how `Cluster` class works ATM, but if it uses `CfnCluster` class underneath, it is only a matter of exposing `kubernetesNetworkConfig?` property.

That's right. But even if the property is not exposed by the high-level construct (`Cluster`, in this case), you can always use an [escape hatch] and do it yourself.

[escape hatch]: https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html

**Comment by Krotki** [comment]
I had a brief look as source code  and correct me if I'm wrong, but I't seems to me that `CfnCluster` class is not used. Instead there is `ClusterResource` which creates `CustomResource` of type `Custom::AWSCDK-EKS-Cluster`. There is a custom resource handler which interprets data from `CustomResource` and creates/updates/deletes resource. I guess only thing missing is to extend handler to handle create (client already supports it) and add error on update. Also extending `CustomResource` with necessary data is necessary and as a last step adding property to `Cluster` class. Am I missing anything ?

I might create a PR for this issue, but testing is a bit of an challenge. Let's say I would like to do some manual tests. How can I install local version of `cdk-eks` locally ? Is there any instructions ?

**Comment by otaviomacedo** [comment]
Yes, take a look at our [contributing guide][1]. In particular, there is a `link-all.sh` script that creates a link from your test application's `node_modules` to your local version of the CDK.

[1]: https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md
